src/main/kotlin/org/partiql/scribe/targets/redshift/RedshiftExcludeTranspilation.md

```diff
@@ -0,0 +1,264 @@
+             // plan representation of `t.flds.b` and `t.flds.c.field_x`
+         )
+      ),
+      projections = listOf(
+          rexStruct(
+              "flds" to rexPathKey(
+                  operand = rexVarRef(...),
+                  key = "flds"
+              ),
+              "foo" to // Rex version of "foo"
+          )
+      )
+   )
+)
+```
+
+Inside the Redshift-specific Rel -> Rel conversion, `RedshiftRewriter`, we remove the `RelExclude` and replace the 
+input to the `RelProject` with the `RelExclude`'s input.
+
+```Kotlin
+RexSelect(
+   input = RelProject( // RelProject's input now points to the RelExclude's input, RelScan
+      input = RelScan( // scan table `EXCLUDE_T`
+         // ...
+      ),
+   ),
+   projections = listOf(
+       rexStruct(
+           "flds" to rexPathKey(
+               operand = rexVarRef(...),
+               key = "flds"
+           ),
+           "foo" to // Rex version of "foo"
+       )
+   )
+)
+```
+
+Looking at the types of the projected nodes, [partiql-lang-kotlin#1764](https://github.com/partiql/partiql-lang-kotlin/pull/1764),
+introduced an additional meta on `PType`s that denotes whether a certain `PType.ROW` or collection type has an excluded
+attribute. For our example, this results in a projection type for the `flds` struct field like the following:
+```Kotlin
+// `flds`'s type
+ROW(
+    fields = listOf(
+        "a": ROW(
+            fields = listOf(
+                "field_x": INTEGER,
+                "field_y": STRING
+            ),
+            metas = emptyMap() // no omitted fields hence no meta
+        ),
+        "c": ROW(
+            fields = listOf(
+                "field_y": STRING
+            ),
+            metas = mapOf("CONTAINS_EXCLUDED_FIELD" = true)
+        )
+    ),
+    metas = mapOf("CONTAINS_EXCLUDED_FIELD" = true)
+),
+```
+
+When the `RedshiftRewriter` then visits a variable reference or path plan node that references a `ROW` field with this
+meta, it will call the `RedshiftExcludeUtils` function `rewriteToObjectTransform`. This utility basically rewrites
+any variable references and paths to use `OBJECT_TRANSFORM` to explicitly include any fields and nested fields we wish
+to keep. Taking the `ROW` type from above from the projection, we will transform that `ROW` type into the following
+`OBJECT_TRANSFORM` call:
+
+```Redshift
+OBJECT_TRANSFORM(
+    t.flds
+    KEEP
+        '"a"',
+        '"c.field_y"'
+)
+```
+Notice how we `KEEP` the path `'"c.field_y"'` rather than `'"c"'` since `c.field_x` was excluded from the query. Also,
+notice how we do not expand `'"a"'` since none of its subfields were excluded. This can help eliminate some redundant
+`KEEP` paths in the query.
+
+The overall query would look something like:
+```Redshift
+SELECT
+    OBJECT_TRANSFORM(
+        "t"."flds" 
+        KEEP
+            '"a"',
+            '"c"."field_y"'
+    ) AS "flds", 
+    "t"."foo" AS "foo"
+FROM "EXCLUDE_T" AS "t"
+```
+
```

Comment:
self-review: this doc doesn't cover the AST rewrites and AST -> text rewrites since they are somewhat more straightforward. I can add more detail on those if that's helpful.